azure/table_azure_monitor_activity_log_event.go

```diff
@@ -0,0 +1,566 @@
+
+	result, err := client.List(ctx, filter, "")
+	if err != nil {
+		plugin.Logger(ctx).Error("azure_monitor_activity_log_event.listMonitorActivityLogEvents", "list", err)
+		return nil, err
+	}
+
+	for _, event := range result.Values() {
+		d.StreamListItem(ctx, event)
+
+		// Check if context has been cancelled or if the limit has been hit (if specified)
+		// if there is a limit, it will return the number of rows required to reach this limit
+		if d.RowsRemaining(ctx) == 0 {
+			return nil, nil
+		}
+	}
+
+	for result.NotDone() {
+		err = result.NextWithContext(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, event := range result.Values() {
+			d.StreamListItem(ctx, event)
+			
+			// Check if context has been cancelled or if the limit has been hit (if specified)
+			// if there is a limit, it will return the number of rows required to reach this limit
+			if d.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+
+	return nil, err
+}
+
+//// UTILITY FUNCTION
+
+func buildActivityLogFilter(quals plugin.KeyColumnQualMap) string {
+	filter := ""
+
+	filterQuals := map[string]string{
+		"event_timestamp":        "eventTimestamp",
+		"resource_group":         "resourceGroupName",
+		"correlation_id":         "correlationId",
+		"resource_id":            "resourceUri",
+		"resource_provider_name": "resourceProvider",
+	}
+
+	for columnName, filterName := range filterQuals {
+
+		/*
+		   The API input must include the Event Timestamp value to filter events. If the user provides an event_timestamp value in the where clause, we use that value. Otherwise, we set it to the past 90 days from the current date because the API can only return events from the last 90 days.
+		*/
+		if columnName == "event_timestamp" && quals[columnName] != nil {
+			for _, q := range quals[columnName].Quals {
+				evTime := q.Value.GetTimestampValue().AsTime().Format(time.RFC3339)
+				if filter == "" {
+					switch q.Operator {
+					case ">", ">=":
+						filter = filterName + " ge " + evTime
+					case "<", "<=":
+						filter = filterName + " le " + evTime
+					}
+				} else {
+					switch q.Operator {
+					case ">", ">=":
+						filter += " and " + filterName + " ge " + evTime
+					case "<", "<=":
+						filter += " and " + filterName + " le " + evTime
+					}
+				}
+			}
+		} else if columnName == "event_timestamp" && quals[columnName] == nil {
+			if filter == "" {
+				filter = filterName + " ge " + time.Now().AddDate(0, 0, -90).Format(time.RFC3339)
+			} else {
+				filter = " and " + filterName + " ge " + time.Now().AddDate(0, 0, -90).Format(time.RFC3339)
+			}
+		} else {
+			if quals[columnName] != nil {
+				for _, q := range quals[columnName].Quals {
+					if q.Operator == "=" {
+						if filter == "" {
+							filter = filterName + " eq " + q.Value.GetStringValue()
+						} else {
+							filter += " and " + filterName + " eq " + q.Value.GetStringValue()
+						}
+					}
+				}
+			}
+		}
+
+	}
+
+	return filter
+}
+
```

Comment:
If we are not keeping the hydrate function we should remove this commented code.